### PR TITLE
Client-side modules should be compilable with jdk8

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

**Comment by bratseth** [comment]
@gjoranv please review

**Comment by bratseth** [comment]
Ping

File: vespa-hadoop/pom.xml [hunk 1]
```diff
@@ -145,8 +155,28 @@
...
+                <configuration>
+                    <jdkToolchain>
+                        <version>${java.version}</version>
+                    </jdkToolchain>
+                    <source>${java.version}</source>
+                    <target>${java.version}</target>
+                    <showWarnings>true</showWarnings>
+                    <optimize>true</optimize>
+                    <showDeprecation>false</showDeprecation>
+                    <compilerArgs>
```
Comment (gjoranv) [inline vespa-hadoop/pom.xml]: AFAICS these are identical to those in 'parent' and can just be inherited from there.  Just remove the whole `compilerArgs`. The same goes for the three options above (showWarnings, optimize, showDeprecation).

File: vespa-http-client/pom.xml [hunk 2]
```diff
@@ -135,8 +158,28 @@
...
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.8.0</version>
+        <configuration>
+          <jdkToolchain>
+            <version>${java.version}</version>
+          </jdkToolchain>
+          <source>${java.version}</source>
+          <target>${java.version}</target>
+          <showWarnings>true</showWarnings>
```
Comment (gjoranv) [inline vespa-http-client/pom.xml]: See comment on the vespa-hadoop pom above. Same goes here.

**Review by gjoranv — CHANGES_REQUESTED** [review]

**Review by gjoranv — APPROVED** [review]

**Pull request merged**